torch/__init__.py

```diff
@@ -270,22 +270,30 @@ def set_default_tensor_type(t):
 
 
 def set_default_dtype(d):
-    r"""Sets the default floating point dtype to :attr:`d`. This type will be
-    used as default floating point type for type inference in
-    :func:`torch.tensor`.
+    r"""Sets the default floating point dtype to :attr:`d`.
+    This type is also used:
+    1. As default floating point type for type inference in :func:`torch.tensor`.
+    2. To determine the default complex dtype. The default complex dtype is set to
+       ``torch.complex128`` if default floating point tensor type is ``torch.float64``,
+       else it's set to ``torch.complex64``
 
     The default floating point dtype is initially ``torch.float32``.
 
     Args:
         d (:class:`torch.dtype`): the floating point dtype to make the default
 
     Example::
-
-        >>> torch.tensor([1.2, 3]).dtype           # initial default for floating point is torch.float32
+        >>> # initial default for floating point is torch.float
```

Comment:
nit: this isn't a huge deal, but I think the previous wording was slightly better, because it matches what is returned for the next line.  As it is now, people will wonder why `torch.float` and  `torch.float32` don't match exactly.

Reply:
ahh ok